File: PizzaTime.Frontend/src/app/data/Order.ts
import { Time } from "@angular/common";
import { Timestamp } from "rxjs";

export enum OrderStatus{
    READY = "ready",
    QUEUED = "queued",
    REFUSED = "refused",
    ACCEPTED = "accepted",
    SERVING = "serving",
    COMPLETED = "completed" ,
    CANCELED = "canceled"
}

export class Order{

    public constructor(
        public id : String,
        public totalPrice: number,
        public date:Date,
        public userId : number,
        public pizzeriaId: String,
        public orderStatus: OrderStatus,
        public orderRows: Array<OrderRow>
    ){}
}


export class OrderRow{
  public constructor(
    public pizzaId : number,
    public baseId : number,
    public  ingredients : Array<number>,
    public quantity : number,
  ){}
}

File: PizzaTime.OrderService/src/main/kotlin/com/PizzaTime/OrderService/Model/Order.kt
package com.PizzaTime.OrderService.Model

import com.google.gson.annotations.Expose
import jakarta.persistence.*
import java.time.Instant
import java.util.*


enum class OrderStatus(val status: String) {
    READY("ready"),
    QUEUED("queued"),
    REFUSED("refused"),
    ACCEPTED("accepted"),
    SERVING("serving"),
    COMPLETED("completed") ,
    CANCELED("canceled")
}


@Entity
@Table(name = "pizza_order")
class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Expose
    var id : String = "";
    @Expose
    var totalPrice: Double = 0.0;

    @Expose
    var address: String? = null;

    @Expose
    var date : Date = Date.from(Instant.now());
    @Expose
    var orderStatus: String = OrderStatus.READY.status;
    @Expose
    var userId : Long = -1;
    @Expose
    var pizzeriaId : Long = -1

    @Expose
    @OneToMany(fetch = FetchType.EAGER)
    var orderRows: Set<OrderRow> = HashSet();
}



